lte/gateway/python/magma/pipelined/rpc_servicer.py

```diff
@@ -975,6 +994,175 @@ def GetStats(self, request, context):
+                    qos_enforce_rule, session_version,
+                )
+            if qos_enforce_rule.ipv6_addr:
+                ipv6 = convert_ipv6_bytes_to_ip_proto(qos_enforce_rule.ipv6_addr)
+                self._ng_deactivate_qer_flows(
+                    ipv6, local_f_teid_ng,
+                    qos_enforce_rule, session_version,
+                )
+
+        elif pdr_entry.pdr_state == PdrState.Value('IDLE'):
+            qos_enforce_rule = pdr_entry.add_qos_enforce_rule
+            if qos_enforce_rule.ip_addr:
+                ipv4 = convert_ipv4_str_to_ip_proto(qos_enforce_rule.ip_addr)
+                self._ng_inactivate_qer_flows(ipv4, qos_enforce_rule, session_version)
+            if qos_enforce_rule.ipv6_addr:
+                ipv6 = convert_ipv6_bytes_to_ip_proto(qos_enforce_rule.ipv6_addr)
+                self._ng_inactivate_qer_flows(ipv6, qos_enforce_rule, session_version)
+
+        # Install PDR rules
+        elif pdr_entry.pdr_state == PdrState.Value('INSTALL'):
+            qos_enforce_rule = pdr_entry.add_qos_enforce_rule
+            if qos_enforce_rule.ip_addr:
+                ipv4 = convert_ipv4_str_to_ip_proto(qos_enforce_rule.ip_addr)
+
+                enforcement_res = \
+                      self._ng_activate_qer_flow(
+                          ipv4, local_f_teid_ng,
+                          qos_enforce_rule, session_version,
+                      )
+                failed_policy_rules_results = \
+                    _retrieve_failed_results(enforcement_res)
+
+            if qos_enforce_rule.ipv6_addr:
+                ipv6 = convert_ipv6_bytes_to_ip_proto(qos_enforce_rule.ipv6_addr)
+                enforcement_res = \
+                      self._ng_activate_qer_flow(
+                          ipv6, local_f_teid_ng,
+                          qos_enforce_rule, session_version,
+                      )
+                failed_policy_rules_results = \
+                    _retrieve_failed_results(enforcement_res)
+
+        return failed_policy_rules_results
+
+    def _ng_inactivate_qer_flows(
+        self, ip: IPAddress, qos_enforce_rule: ActivateFlowsRequest,
+        session_version,
+    ):
+
+        self._ng_update_version(
+            qos_enforce_rule.sid.id, qos_enforce_rule,
+            ip, session_version,
+        )
+
+        rule_ids = [policy.rule.id for policy in qos_enforce_rule.policies]
+
+        if rule_ids:
+            self._enforcer_app.deactivate_rules(
+                qos_enforce_rule.sid.id, ip,
+                rule_ids,
+            )
+
+    def _ng_activate_qer_flow(
+        self, ip: IPAddress, local_f_teid_ng,
+        qos_enforce_rule: ActivateFlowsRequest, session_version,
+    ):
+
+        self._ng_update_version(
+            qos_enforce_rule.sid.id, qos_enforce_rule,
+            ip, session_version,
+        )
+
+        # Install rules in enforcement
+        enforcement_res = self._activate_rules_in_enforcement(
```

Comment:
from here until the end of this function its essentially running _install_flows_gx (if you remove the self._update_version(request, ip_address) from it)

Can we just cut out the `_update_version` call from the `_install_flows_gx` function and that way we can reuse it here as well

Reply:
Yes, We modified it. Please review it.